Library/Homebrew/version.rb

```diff
@@ -179,6 +179,18 @@ def self.detect(url, specs)
     end
   end
 
+  def self.create(val)
+    if val.respond_to?(:to_str)
+      if val.start_with?("HEAD")
+        HeadVersion.new(val)
+      else
+        Version.new(val)
+      end
+    else
+      raise TypeError, "Version value must be a string; got a #{val.class} (#{val})"
+    end
+  end
```

Comment:
This is purely a style issue, but here's a suggestion that is easier to read in my opinion:

```ruby
def self.create(val)
  unless val.respond_to?(:to_str)
    raise TypeError, "Version value must be a string; got a #{val.class} (#{val})"
  end

  if val.start_with?("HEAD")
    HeadVersion.new(val)
  else
    Version.new(val)
  end
end
```

An even more radical version would be to replace the final `if` statement with (but I'm not sure this is considered good style and would only adopt it if other maintainers agree):

```ruby
klass = val.start_with?("HEAD") ? HeadVersion : Version
klass.new(val)
```